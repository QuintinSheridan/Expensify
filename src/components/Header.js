import React from 'react'
import {NavLink} from 'react-router-dom'

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to='/' exact={true} activeStyle={{
            fontWeight: "bold",
            color: "red"
          }}> Home </NavLink>
        <NavLink to='/create' activeStyle={{
            fontWeight: "bold",
            color: "red"
          }}> Create </NavLink>
        <NavLink to='/help' activeStyle={{
            fontWeight: "bold",
            color: "red"
          }}> Help </NavLink>
    </header>
);


export default Header;