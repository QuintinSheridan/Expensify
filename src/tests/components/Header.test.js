// react-test-render
import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Header from '../../components/Header';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
// shallow rendering renders components while full rendering will render the children components as well

test("should render Header correctly", () => {
    const wrapper = shallow(<Header />)
    expect(toJSON(wrapper)).toMatchSnapshot()
    expect(wrapper.find('h1').length).toBe(1)
})