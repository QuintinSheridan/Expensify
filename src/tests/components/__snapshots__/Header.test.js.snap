// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeStyle={
      Object {
        "color": "red",
        "fontWeight": "bold",
      }
    }
    exact={true}
    to="/"
  >
     Home 
  </NavLink>
  <NavLink
    activeStyle={
      Object {
        "color": "red",
        "fontWeight": "bold",
      }
    }
    to="/create"
  >
     Create 
  </NavLink>
  <NavLink
    activeStyle={
      Object {
        "color": "red",
        "fontWeight": "bold",
      }
    }
    to="/help"
  >
     Help 
  </NavLink>
</header>
`;
